### PR TITLE
[KOA-4578]: Adopting new Backpack foundations packages

### PR DESCRIPTION
**Pull request by olliecurtis**
This PR adopts the new foundations packages and all the required changes to use the correct packages, where necessary.

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [ ] `README.md`
+ [ ] Tests
+ [ ] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/main/Example/Backpack%20Screenshot/Screenshots.swift)
+ [ ] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/main/Backpack/Backpack.h)
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

**Comment by k0nserv** [comment]
This looks good to me. You need to re-generate the icon snapshot tests to make CI pass. I can either do this for you or walk you through how to do it?

File: UNRELEASED.md [hunk 1]
```diff
@@ -2,6 +2,12 @@
 
 > Place your changes below this line.
 
+**Changed:**
```
Comment (k0nserv) [inline UNRELEASED.md:5]: This is non-breaking to consumers, right?

`Changed` usually indicates a breaking change. We need to make sure we don't release this as breaking as a result

Comment (olliecurtis) [reply UNRELEASED.md:5]: no this change is not breaking. My apologies, I thought changed is minor and `Breaking` is breaking?

Comment (k0nserv) [reply UNRELEASED.md]: We've historically used "Changed" to mean breaking i.e. something has changed and the user will have to modify how they are using Backpack in response. At least that's how I remember things

**Review by k0nserv — APPROVED** [review]

**Pull request merged**